urlarbiter/url_arbiter.go

```diff
@@ -57,14 +58,17 @@ func (u *URLArbiter) Register(path, publishingAppName string) (URLArbiterRespons
 		return URLArbiterResponse{}, err
 	}
 
-	// Read the response body and then check the status code so we can
-	// return the errors from the response.
-	switch response.StatusCode {
-	case 422: // Unprocessable Entity.
-		return arbiterResponse, UnprocessableEntity
-	case http.StatusConflict:
-		return arbiterResponse, ConflictPathAlreadyReserved
+	if response.StatusCode < 300 && response.StatusCode >= 200 {
+		// 1xx and 2xx
+		return arbiterResponse, nil
+	} else {
+		switch response.StatusCode {
+		case 422: // Unprocessable Entity.
+			return arbiterResponse, UnprocessableEntity
+		case http.StatusConflict:
+			return arbiterResponse, ConflictPathAlreadyReserved
+		default:
+			return arbiterResponse, UnexpectedResponse
```

Comment:
as i read further, i wonder whether moving the `if` condition above into this `default` block would read better?